src/debugging/python/PythonDebugHelper.ts

```diff
@@ -0,0 +1,196 @@
+    }
+
+    public async resolveDebugConfiguration(context: DockerDebugContext, debugConfiguration: PythonDockerDebugConfiguration): Promise<ResolvedDebugConfiguration | undefined> {
+        const containerName = inferContainerName(debugConfiguration, context, context.folder.name);
+
+        // Since Python is a special case, we need to ensure the container is removed before attempting to resolve
+        // the debug configuration.
+        try {
+            await this.cliDockerClient.removeContainer(containerName, { force: true });
+        } catch {}
+
+        const debuggerLogFilePath = PythonExtensionHelper.getDebuggerLogFilePath(context.folder.name);
+        await fse.remove(debuggerLogFilePath);
+
+        let debuggerReadyPromise = new Promise((resolve) => resolve());
+        if (debugConfiguration.preLaunchTask) {
+            // There is this limitation with the Python debugger where we need to ensure it's ready before allowing VSCode to attach,
+            // if attach happens too soon then it will fail silently. The workaround here is to set the preLaunchTask to undefined,
+            // then execute it ourselves with a listener to when it is finished, then wait for the debugger to be ready and return
+            // the resolved launch configuration.
+
+            const task = await this.tryGetPrelaunchTask(debugConfiguration.preLaunchTask);
+
+            if (!task) {
+                throw new Error(`Unable to find the prelaunch task with the name: ${debugConfiguration.preLaunchTask}`);
+            }
+
+            debugConfiguration.preLaunchTask = undefined;
+
+            /* eslint-disable-next-line @typescript-eslint/no-floating-promises */
+            vscode.tasks.executeTask(task);
+
+            debuggerReadyPromise = PythonExtensionHelper.ensureDebuggerReady(task, debuggerLogFilePath, containerName, this.cliDockerClient);
+        }
+
+        return await debuggerReadyPromise.then(() => {
+            return this.resolveDebugConfigurationInternal(debugConfiguration, containerName, context);
+        });
+    }
+
+    private resolveDebugConfigurationInternal(debugConfiguration: PythonDockerDebugConfiguration, containerName: string, context: DockerDebugContext): ResolvedDebugConfiguration {
+        const projectType = debugConfiguration.python.projectType;
+
+        const dockerServerReadyAction =
+      resolveDockerServerReadyAction(
+          debugConfiguration,
+          {
+              containerName: containerName,
+              pattern: this.getServerReadyPattern(projectType),
+              uriFormat: "%s://localhost:%s"
+          },
+          true);
+
+        // These properties are required by the old debugger, should be changed to normal properties in the configuration
+        // as soon as the new debugger is released to 100% of the users.
+        const debugOptions = ["FixFilePathCase", "RedirectOutput", "ShowReturnValue"];
+
+        if (this.localOsProvider.os === "Windows") {
```

Comment:
Since `localOsProvider` and most of its siblings will eventually be removed or refactored, I used the `os` package, along with `os.platform() === 'win32'` etc. for most of these checks. Should we do the same here?

Reply:
Sure, if it will be removed then yeah it makes sense.